src/mcp/client/auth/oauth2.py

```diff
@@ -96,6 +99,7 @@ class OAuthContext:
     redirect_handler: Callable[[str], Awaitable[None]] | None
     callback_handler: Callable[[], Awaitable[tuple[str, str | None]]] | None
     timeout: float = 300.0
+    client_metadata_url: str | None = None  # SEP-991: URL-based client ID
```

Comment:
nit: personally think we shouldn't litter comments with references to SEP numbers, hard to keep up-to-date.